### DISCUSSION TITLE
No Endpoint to Verify Email?

### DISCUSSION THREAD

**Discussion by osseonews** [nhost/nhost#2730] [Q&A]
I didn't want to post an issue on this, as I'm not sure if I'm missing something or not. But, I looked thru all of the documentation and I can't find one endpoint to actually verify the email. The only way to do this, is to have a redirect to nhost.auth.url}/verify. This becomes a complicated mess though, because of the following:

1. In order to do this correctly, you need to set up the redirect in your app, as explained here: https://nhost.io/blog/protect-magic-links-from-email-clients. So this adds another redirect to the equation. You redirect the user to your app, and then they get redirected to nhost.auth.url and then back to your app. So now 3 redirects.

2. When you verify the email, you have no control whatsoever for the redirect URL, and it will always redirect initially to the Client URL in the Authentication Settings in the nhost panel. But, why is there is only one Client URL? And why do we need to redirect to that? Attaching a redirectUrl to the initial nhost.auth.url will cause the app to redirect first to the Client URL and then to the final destination URL.  So then you are talking about 4 redirects. 

3. When using the Nhost provider, when you verify the user email, the user is immediately logged in? This is strange behavior, because sometimes you want to verify the email first and then have the user log in. Essentially, the verify email acts as a passwordless login, but then why do you need a passwordless login, just send a verify email to the user and they are logged in?

Overall, the entire process is very strange and even in the worst case, you are looking at 3 redirects.

My suggestion is to simply create a backend POST endpoint that verifies the email. If this exists already, then great, can you please share it? If not, just create one, so you can send a POST request to that endpoint with the verify TOKEN provided by nhost in body and the endpoint sends back a success or failure with an error message. This is what is done now presumably on the nhost servers themsevles, but it's only done on the Nhost server, and is not exposed, so there is no way to call that function directly as far as I can tell. So you got to go thru this confusing mess of redirects and unintended functionality (logging in user automatically).

If you just had a simple POST request to verify an email, then the process is easy. You get the TOKEN, you send to the endpoint, you get back a response. Based on that response, you can then do as you wish on the frontend. No need for automated redirects or anything.

**Comment by dbarrosop** [comment]
Adding a POST-equivalent endpoint for convenience is a good idea (would you mind opening a github issue to track it?). In the meantime, though, you should be able to use the same GET endpoint we already have yourself for this. If you fetch it from javascript you will get a 302 REDIRECT but the redirect won't happen automatically. Then you can parse the redirect URL you got and extract the refresh token or the error if any, so functionality wise it should be the same as the POST, just a bit less convenient.

**Reply by osseonews** [reply]
OK, will try the GET Request. Also, what's the difference between verifying the email via the endpoint or do a MagicLink? Both flows send back the access token and refresh token, so can't you use useSendVerificationEmail to send the verification email to a customer instead of the Magic Link via useSignInEmailPasswordless()? Seems like these do the same exact thing.

**Reply by osseonews** [reply]
Sorry, I tried a Get request to the nhost.auth.url}/verify and it doesn't do anything or return anything. It just redirects to the Client URL. Can you please post what the GET request actually is supposed to be to verify an email?

**Reply by osseonews** [reply]
I noticed also that when you send the PasswordReset link this also logs in the customer on the frontend, when using the nHostprovider, even before they reset the password. So this is exactly like passwordless log in also. It's a bit confusing that all these emails with redirects to verify basically just do the same exact thing.
